toolkit/tools/internal/pkgjson/pkgjson_test.go

```diff
@@ -257,25 +284,116 @@ func TestSubsetValidOutside(t *testing.T) {
+		existingPkgVer1, existingPkgCond1, existingPkgVer2, existingPkgCond2 string
+		queryPkgVer1, queryPkgCond1, queryPkgVer2, queryPkgCond2             string
+		expected                                                             bool
+	}{
+		{
+			name:            "p < q, unbounded",
+			existingPkgVer1: "1", existingPkgCond1: ">",
+			queryPkgVer1: "3", queryPkgCond1: ">",
+			expected: false,
+		},
+		{
+			name:            "p < q, bounded",
+			existingPkgVer1: "1", existingPkgCond1: ">", existingPkgVer2: "5", existingPkgCond2: "<",
+			queryPkgVer1: "3", queryPkgCond1: ">",
+			expected: false,
+		},
+		{
+			name:            "p <= q, unbounded",
+			existingPkgVer1: "1", existingPkgCond1: ">",
+			queryPkgVer1: "1", queryPkgCond1: ">",
+			expected: true,
+		},
+		{
+			name:            "p <= q, bounded",
+			existingPkgVer1: "1", existingPkgCond1: ">", existingPkgVer2: "5", existingPkgCond2: "<",
+			queryPkgVer1: "1", queryPkgCond1: ">",
+			expected: true,
+		},
+		{
+			name:            "p > q, unbounded",
+			existingPkgVer1: "5", existingPkgCond1: "<",
+			queryPkgVer1: "1", queryPkgCond1: "<",
+			expected: false,
+		},
+		{
+			name:            "p > q, bounded",
+			existingPkgVer1: "5", existingPkgCond1: "<", existingPkgVer2: "1", existingPkgCond2: ">",
+			queryPkgVer1: "1", queryPkgCond1: "<",
+			expected: false,
+		},
+		{
+			name:            "p >= q, unbounded",
+			existingPkgVer1: "5", existingPkgCond1: "<",
+			queryPkgVer1: "5", queryPkgCond1: "<",
+			expected: true,
+		},
+		{
+			name:            "p >= q, bounded",
+			existingPkgVer1: "5", existingPkgCond1: "<", existingPkgVer2: "1", existingPkgCond2: ">",
+			queryPkgVer1: "5", queryPkgCond1: "<",
+			expected: true,
+		},
+		{
+			name:            "p = q, unbounded",
+			existingPkgVer1: "1", existingPkgCond1: ">",
+			queryPkgVer1: "1", queryPkgCond1: ">",
+			expected: true,
+		},
+		{
+			name:            "p = q, one bounded",
+			existingPkgVer1: "1", existingPkgCond1: ">", existingPkgVer2: "5", existingPkgCond2: "<",
+			queryPkgVer1: "1", queryPkgCond1: ">",
+			expected: true,
+		},
+		{
+			name:            "p = q, both bounded",
+			existingPkgVer1: "1", existingPkgCond1: ">", existingPkgVer2: "5", existingPkgCond2: "<",
+			queryPkgVer1: "1", queryPkgCond1: ">", queryPkgVer2: "5", queryPkgCond2: "<",
+			expected: true,
+		},
+		{
+			name:            "subset",
+			existingPkgVer1: "1", existingPkgCond1: ">", existingPkgVer2: "3", existingPkgCond2: "<",
+			queryPkgVer1: "1", queryPkgCond1: ">=", queryPkgVer2: "5", queryPkgCond2: "<=",
+			expected: true,
+		},
+	}
+	for _, test := range testInputs {
+		t.Run(test.name, func(t *testing.T) {
+			p1 := &PackageVer{Version: test.existingPkgVer1, Condition: test.existingPkgCond1, SVersion: test.existingPkgVer2, SCondition: test.existingPkgCond2}
+			p2 := &PackageVer{Version: test.queryPkgVer1, Condition: test.queryPkgCond1, SVersion: test.queryPkgVer2, SCondition: test.queryPkgCond2}
```

Comment:
Could we stick to one naming scheme, so it's easier to follow the logic? Right now we have `existing` vs `query`, `p1` vs `p2`, and `p` vs `q` all at once and it's making this a bit confusing.